en/02_Developer_Guides/09_Security/04_Sudo_Mode.md

```diff
@@ -6,52 +6,69 @@ icon: key
 
-Sudo mode will automatically be enabled for the configured lifetime when a user logs into the CMS. Note that if the PHP session lifetime expires before the sudo mode lifetime, that sudo mode will also be cleared (and re-enabled when the user logs in again). If the user leaves their CMS open, or continues to use it, for an extended period of time with automatic refreshing in the background, sudo mode will eventually deactivate once the max lifetime is reached.
+Sudo mode is not active when a user logs in to the CMS. If the PHP session lifetime expires before the sudo mode lifetime, that sudo mode will also be cleared. If the user leaves their CMS open, or continues to use it, for an extended period of time with automatic refreshing in the background, sudo mode will eventually deactivate once the max lifetime is reached.
 
-## Configuring the lifetime
-
-The default [`SudoModeServiceInterface`](api:SilverStripe\Security\SudoMode\SudoModeServiceInterface) implementation is [`SudoModeService`](api:SilverStripe\Security\SudoMode\SudoModeService), and its lifetime can be configured with YAML. You should read the lifetime value using `SudoModeServiceInterface::getLifetime()`.
+Data protected by sudo mode is still viewable without entering the password, but any actions that modify data will require the user to enter their password. This is done to balance usability with security. The following [`DataObject`](api:SilverStripe\ORM\DataObject) subclasses are protected by sudo mode:
 
-```yml
-SilverStripe\Security\SudoMode\SudoModeService:
-  lifetime_minutes: 25
-```
+- [`Member`](api:SilverStripe\Security\Member)
+- [`Group`](api:SilverStripe\Security\Group)
+- [`PermissionRole`](api:SilverStripe\Security\PermissionRole)
+- [`PermissionRoleCode`](api:SilverStripe\Security\PermissionRoleCode)
+- [`SiteConfig`](api:SilverStripe\SiteConfig\SiteConfig)
 
-## Enabling sudo mode for controllers
+## Adding or removing sudo mode for a `DataObject`
 
-You can add the `SudoModeServiceInterface` singleton as a dependency to a controller that requires sudo mode for one of its actions:
+To add sudo mode for a particular `DataObject`, simply set the [`DataObject.require_sudo_mode`](api:SilverStripe\ORM\DataObject->require_sudo_mode) configuration property to `true`, either directly on the class or via yml.
 
 ```php
-namespace App\Control;
+namespace App\Model;
 
-class MyController extends Controller
+use SilverStripe\ORM\DataObject;
+
+class Product extends DataObject
 {
-    private ?SudoModeServiceInterface $sudoModeService = null;
+    // ...
+    private static bool $require_sudo_mode = true;
+}
+```
+
+```yml
+SomeModule\Model\Player:
+  require_sudo_mode: true
+```
 
-    private static array $dependencies = ['SudoModeService' => '%$' . SudoModeServiceInterface::class];
+> [!NOTE]
+> This will only add sudo mode to `DataObject` edit forms within the CMS interface. It will have no effect on forms outside of the CMS, such as custom forms in the frontend.
```

Comment:
```suggestion
> This will only add sudo mode to edit forms within the CMS interface. It will have no effect on forms outside of the CMS, such as custom forms in the frontend.
```

Reply:
Updated